### ISSUE TITLE
Format of macro classes such as SFERES_FITNESS

### ISSUE THREAD

**Issue by vassilisvas** [resibots/format_code#3]
I don't like the way the atom clang-format package formats macros such as SFERES_FITNESS.

For example, the following function is formatted in a nicely looking way:

template <typename Indiv> void eval(const Indiv &ind) {
  float v = 0;
  std::vector<double> values = {0.2, 0.5, 0.3};

  for (size_t i = 0; i < 3; ++i) {
    std::cout << "bla ";
  }

  this->_value = v;
}

..but being inside SFERES_FITNESS the result is the following:

SFERES_FITNESS(SomeFunction, sferes::fit::Fitness){
  public : template <typename Indiv> void eval(const Indiv &ind){float v = 0;
std::vector<double> values = {0.2, 0.5, 0.3};

for (size_t i = 0; i < 3; ++i) {
  std::cout << "bla ";
}

this->_value = v;
}
}
;

Do you agree?

**Comment by costashatz** [comment]
I agree with you. On the other hand, I don't think there's a way to handle these cases very well for the moment. What I've been doing is the following:

``` cpp

// clang-format off
SFERES_FITNESS(SomeFunction, sferes::fit::Fitness){
// clang-format on

// rest of code

// clang-format off
};
// clang-format on
```
